### ISSUE TITLE
[C++20][Clang] frontend error

### ISSUE THREAD

**Issue by SAtacker** [llvm/llvm-project#60380]
Repository: https://github.com/SAtacker/quick-ftxui

Terminal Output: `cmake .. -DCMAKE_CXX_STANDARD=20`

```
satacker@ubuntu:/home/extended/quick-ftxui/build$ ninja
[0/1] Re-running CMake...
-- clang-tidy not found.
-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE) 
Doxygen need to be installed to generate the doxygen documentation
-- Configuring done
-- Generating done
-- Build files have been written to: /home/extended/quick-ftxui/build
[12/66] Building CXX object _deps/ftxui-build/CMakeFiles/dom.dir/src/ftxui/dom/border.cpp.o
FAILED: _deps/ftxui-build/CMakeFiles/dom.dir/src/ftxui/dom/border.cpp.o 
/usr/bin/c++  -I/home/extended/quick-ftxui/build/_deps/ftxui-src/include -I/home/extended/quick-ftxui/build/_deps/ftxui-src/src -g -Wall -Wextra -pedantic -Werror -Wmissing-declarations -Wdeprecated -Wshadow -std=gnu++20 -MD -MT _deps/ftxui-build/CMakeFiles/dom.dir/src/ftxui/dom/border.cpp.o -MF _deps/ftxui-build/CMakeFiles/dom.dir/src/ftxui/dom/border.cpp.o.d -o _deps/ftxui-build/CMakeFiles/dom.dir/src/ftxui/dom/border.cpp.o -c /home/extended/quick-ftxui/build/_deps/ftxui-src/src/ftxui/dom/border.cpp
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace, preprocessed source, and associated run script.
Stack dump:
0.	Program arguments: /usr/bin/c++ -I/home/extended/quick-ftxui/build/_deps/ftxui-src/include -I/home/extended/quick-ftxui/build/_deps/ftxui-src/src -g -Wall -Wextra -pedantic -Werror -Wmissing-declarations -Wdeprecated -Wshadow -std=gnu++20 -MD -MT _deps/ftxui-build/CMakeFiles/dom.dir/src/ftxui/dom/border.cpp.o -MF _deps/ftxui-build/CMakeFiles/dom.dir/src/ftxui/dom/border.cpp.o.d -o _deps/ftxui-build/CMakeFiles/dom.dir/src/ftxui/dom/border.cpp.o -c /home/extended/quick-ftxui/build/_deps/ftxui-src/src/ftxui/dom/border.cpp
1.	/home/extended/quick-ftxui/build/_deps/ftxui-src/src/ftxui/dom/border.cpp:25:2: current parser token ';'
2.	/home/extended/quick-ftxui/build/_deps/ftxui-src/src/ftxui/dom/border.cpp:14:1: parsing namespace 'ftxui'
 #0 0x00007f7a1f6d1d01 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/lib/x86_64-linux-gnu/libLLVM-14.so.1+0xe3fd01)
 #1 0x00007f7a1f6cfa3e llvm::sys::RunSignalHandlers() (/lib/x86_64-linux-gnu/libLLVM-14.so.1+0xe3da3e)
 #2 0x00007f7a1f6d10ab llvm::sys::CleanupOnSignal(unsigned long) (/lib/x86_64-linux-gnu/libLLVM-14.so.1+0xe3f0ab)
 #3 0x00007f7a1f5fddff (/lib/x86_64-linux-gnu/libLLVM-14.so.1+0xd6bdff)
 #4 0x00007f7a1e379520 (/lib/x86_64-linux-gnu/libc.so.6+0x42520)
 #5 0x00007f7a25e99e23 (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0xd35e23)
 #6 0x00007f7a25e93b45 (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0xd2fb45)
 #7 0x00007f7a25e47b71 (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0xce3b71)
 #8 0x00007f7a25e9a973 (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0xd36973)
 #9 0x00007f7a25e93b45 (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0xd2fb45)
#10 0x00007f7a25e97841 (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0xd33841)
#11 0x00007f7a25e47b71 (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0xce3b71)
#12 0x00007f7a25e9a973 (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0xd36973)
#13 0x00007f7a25e93b45 (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0xd2fb45)
#14 0x00007f7a25e47b71 (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0xce3b71)
#15 0x00007f7a25e92def (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0xd2edef)
#16 0x00007f7a25e47b47 (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0xce3b47)
#17 0x00007f7a25e984cd (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0xd344cd)
#18 0x00007f7a25e47b71 (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0xce3b71)
#19 0x00007f7a25e92e87 (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0xd2ee87)
#20 0x00007f7a25e47b47 (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0xce3b47)
#21 0x00007f7a25e984cd (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0xd344cd)
#22 0x00007f7a25e97841 (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0xd33841)
#23 0x00007f7a25e47b71 (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0xce3b71)
#24 0x00007f7a25e48488 clang::Expr::EvaluateAsInitializer(clang::APValue&, clang::ASTContext const&, clang::VarDecl const*, llvm::SmallVectorImpl<std::pair<clang::SourceLocation, clang::PartialDiagnostic> >&, bool) const (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0xce4488)
#25 0x00007f7a25dbaae4 clang::VarDecl::evaluateValueImpl(llvm::SmallVectorImpl<std::pair<clang::SourceLocation, clang::PartialDiagnostic> >&, bool) const (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0xc56ae4)
#26 0x00007f7a25dbac95 clang::VarDecl::checkForConstantInitialization(llvm::SmallVectorImpl<std::pair<clang::SourceLocation, clang::PartialDiagnostic> >&) const (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0xc56c95)
#27 0x00007f7a263f3bbd clang::Sema::CheckCompleteVariableDeclaration(clang::VarDecl*) (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0x128fbbd)
#28 0x00007f7a263f288b clang::Sema::AddInitializerToDecl(clang::Decl*, clang::Expr*, bool) (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0x128e88b)
#29 0x00007f7a25b83c90 clang::Parser::ParseDeclarationAfterDeclaratorAndAttributes(clang::Declarator&, clang::Parser::ParsedTemplateInfo const&, clang::Parser::ForRangeInit*) (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0xa1fc90)
#30 0x00007f7a25b80e3f clang::Parser::ParseDeclGroup(clang::ParsingDeclSpec&, clang::DeclaratorContext, clang::SourceLocation*, clang::Parser::ForRangeInit*) (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0xa1ce3f)
#31 0x00007f7a25c285ab clang::Parser::ParseDeclOrFunctionDefInternal(clang::ParsedAttributesWithRange&, clang::ParsingDeclSpec&, clang::AccessSpecifier) (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0xac45ab)
#32 0x00007f7a25c28060 clang::Parser::ParseDeclarationOrFunctionDefinition(clang::ParsedAttributesWithRange&, clang::ParsingDeclSpec*, clang::AccessSpecifier) (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0xac4060)
#33 0x00007f7a25c274da clang::Parser::ParseExternalDeclaration(clang::ParsedAttributesWithRange&, clang::ParsingDeclSpec*) (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0xac34da)
#34 0x00007f7a25b95cbc clang::Parser::ParseInnerNamespace(llvm::SmallVector<clang::Parser::InnerNamespaceInfo, 4u> const&, unsigned int, clang::SourceLocation&, clang::ParsedAttributes&, clang::BalancedDelimiterTracker&) (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0xa31cbc)
#35 0x00007f7a25b95809 clang::Parser::ParseNamespace(clang::DeclaratorContext, clang::SourceLocation&, clang::SourceLocation) (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0xa31809)
#36 0x00007f7a25b7b88e clang::Parser::ParseDeclaration(clang::DeclaratorContext, clang::SourceLocation&, clang::ParsedAttributesWithRange&, clang::SourceLocation*) (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0xa1788e)
#37 0x00007f7a25c26c74 clang::Parser::ParseExternalDeclaration(clang::ParsedAttributesWithRange&, clang::ParsingDeclSpec*) (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0xac2c74)
#38 0x00007f7a25c25b1d clang::Parser::ParseTopLevelDecl(clang::OpaquePtr<clang::DeclGroupRef>&, bool) (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0xac1b1d)
#39 0x00007f7a25b688ed clang::ParseAST(clang::Sema&, bool, bool) (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0xa048ed)
#40 0x00007f7a26cdcb71 clang::CodeGenAction::ExecuteAction() (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0x1b78b71)
#41 0x00007f7a27678b57 clang::FrontendAction::Execute() (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0x2514b57)
#42 0x00007f7a275d03a6 clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0x246c3a6)
#43 0x00007f7a276f245b clang::ExecuteCompilerInvocation(clang::CompilerInstance*) (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0x258e45b)
#44 0x000000000041328b cc1_main(llvm::ArrayRef<char const*>, char const*, void*) (/usr/bin/c+++0x41328b)
#45 0x00000000004114bc (/usr/bin/c+++0x4114bc)
#46 0x00007f7a2724eed2 (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0x20eaed2)
#47 0x00007f7a1f5fdb6d llvm::CrashRecoveryContext::RunSafely(llvm::function_ref<void ()>) (/lib/x86_64-linux-gnu/libLLVM-14.so.1+0xd6bb6d)
#48 0x00007f7a2724e9c0 clang::driver::CC1Command::Execute(llvm::ArrayRef<llvm::Optional<llvm::StringRef> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, bool*) const (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0x20ea9c0)
#49 0x00007f7a27219183 clang::driver::Compilation::ExecuteCommand(clang::driver::Command const&, clang::driver::Command const*&) const (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0x20b5183)
#50 0x00007f7a2721940a clang::driver::Compilation::ExecuteJobs(clang::driver::JobList const&, llvm::SmallVectorImpl<std::pair<int, clang::driver::Command const*> >&) const (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0x20b540a)
#51 0x00007f7a27233507 clang::driver::Driver::ExecuteCompilation(clang::driver::Compilation&, llvm::SmallVectorImpl<std::pair<int, clang::driver::Command const*> >&) (/lib/x86_64-linux-gnu/libclang-cpp.so.14+0x20cf507)
#52 0x0000000000410f26 main (/usr/bin/c+++0x410f26)
#53 0x00007f7a1e360d90 __libc_start_call_main ./csu/../sysdeps/nptl/libc_start_call_main.h:58:16
#54 0x00007f7a1e360e40 call_init ./csu/../csu/libc-start.c:128:20
#55 0x00007f7a1e360e40 __libc_start_main ./csu/../csu/libc-start.c:379:5
#56 0x000000000040e3b5 _start (/usr/bin/c+++0x40e3b5)
clang: error: clang frontend command failed with exit code 139 (use -v to see invocation)
Ubuntu clang version 14.0.0-1ubuntu1
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
clang: note: diagnostic msg: 
********************

PLEASE ATTACH THE FOLLOWING FILES TO THE BUG REPORT:
Preprocessed source(s) and associated run script(s) are located at:
clang: note: diagnostic msg: /tmp/border-d25d23.cpp
clang: note: diagnostic msg: /tmp/border-d25d23.sh
clang: note: diagnostic msg: 

********************
[21/66] Building CXX object CMakeFiles/quick-ftxui.dir/src/quick-ftxui.cpp.o
ninja: build stopped: subcommand failed.
```

Files : /tmp/border-d25d23.cpp and /tmp/border-d25d23.sh can be found https://gist.github.com/SAtacker/fdcbec1df80a099eb0a610747fe3cdf3

Thanks!

**Comment by SAtacker** [comment]
Even for clang++-12 it crashes. The CI/CD logs can be found in this run https://github.com/SAtacker/quick-ftxui/actions/runs/4042568667/jobs/6950426757#step:5:51

**Comment by DimitryAndric** [comment]
Duplicate of #56016, which was fixed in d1942855c4317c61f9fae173afa2cbe1076c3c4c ("[clang] Consider array filler in MaybeElementDependentArrayfiller()") by @tbaederr.
